cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -3569,6 +3569,65 @@ pAnchorDataHash =
       , Opt.help "Proposal anchor data hash (obtain it with \"cardano-cli hash anchor-data ...\")"
       ]
 
+data HashCheckParamInfo anchorData
+  = HashCheckParamInfo
+  { flagSuffix :: String
+  , dataName :: String
+  , hashParamName :: String
+  , urlParamName :: String
+  }
+
+pMustCheckHash :: HashCheckParamInfo anchorData -> Parser (MustCheckHash anchorData)
+pMustCheckHash
+  ( HashCheckParamInfo
+      { flagSuffix = flagSuffix'
+      , dataName = dataName'
+      , hashParamName = hashParamName'
+      , urlParamName = urlParamName'
+      }
+    ) =
+    asum
+      [ Opt.flag' CheckHash $
+          mconcat
+            [ Opt.long ("check-" ++ flagSuffix')
+            , Opt.help
+                ( "Check the "
+                    ++ dataName'
+                    ++ " hash (from "
+                    ++ hashParamName'
+                    ++ ") by downloading "
+                    ++ dataName'
+                    ++ " data (from "
+                    ++ urlParamName'
+                    ++ ")."
+                )
+            ]
+      , Opt.flag' TrustHash $
+          mconcat
+            [ Opt.long ("trust-" ++ flagSuffix')
+            , Opt.help
+                ("Do not check the " ++ dataName' ++ " hash (from " ++ hashParamName' ++ ") and trust it is correct.")
+            ]
+      ]
+
+proposalHashCheckInfo :: HashCheckParamInfo ProposalUrl
```

Comment:
Please turn these into stand alone parsers. You can create a function that creates a parser which is the usual style we have.